lib/HttpClient.js

```diff
@@ -215,17 +206,8 @@ function rawRequest(opts, cb) {
             // this first before we abort the request so we can pick up socket
             // information like the ip.
             var err = errors.createConnectTimeoutErr(opts, req);
-
-            if (req) {
-                req.abort();
-            }
-
-            dtrace._rstfy_probes['client-error'].fire(function () {
-                return ([id, err.toString()]);
-            });
-            cb(err, req);
-            emitResult(err, req, null);
-            emitAfter(req, null, err);
+            req._forcedAbortErr = err;
```

Comment:
General question: what about using a symbol here to avoid stomping on consumer's properties?

Reply:
@hekike and I were chatting about this the other day - we'd like to add an "engines" field into package.json so we can begin to safely use ES6 stuff, but that'd be reserved for a breaking change. This isn't ideal, but it would keep us from breaking any existing contracts.

Reply:
Absolutely, that makes sense, so this looks good to me!